test-tests/shared/src/test/scala/zio/test/DefaultTestReporterSpec.scala

```diff
@@ -110,35 +107,7 @@ object DefaultTestReporterSpec extends AsyncBaseSpec {
     ) + "\n"
   }
 
-  def reportSuccess =
-    check(test1, Vector(test1Expected, reportStats(1, 0, 0)))
-
-  def reportFailure =
-    check(test3, test3Expected :+ reportStats(0, 0, 1))
-
-  def reportError =
-    check(test4, test4Expected :+ reportStats(0, 0, 1))
-
-  def reportSuite1 =
-    check(suite1, suite1Expected :+ reportStats(2, 0, 0))
-
-  def reportSuite2 =
-    check(suite2, suite2Expected :+ reportStats(2, 0, 1))
-
-  def reportSuites =
-    check(
-      suite("Suite3")(suite1, test3),
-      Vector(expectedFailure("Suite3")) ++ suite1Expected.map(withOffset(2)) ++ test3Expected
-        .map(withOffset(2)) :+ reportStats(2, 0, 1)
-    )
-
-  def simpleAssertion =
-    check(test5, test5Expected :+ reportStats(0, 0, 1))
-
-  def multipleNestedFailures =
-    check(test6, test6Expected :+ reportStats(0, 0, 1))
-
-  def check[E](spec: ZSpec[TestEnvironment, String, String, Unit], expected: Vector[String]): Future[Boolean] =
+  def check[E](spec: ZSpec[TestEnvironment, String, String, Unit], expected: Vector[String]) =
```

Comment:
Eliminate the `unsafeRunWith` and just have this return a `ZIO`. Delete the `unsafeRunWith` method below. Don't take a parameter for the expected result and instead just have the `ZIO` return the actual output. Rename so people don't confuse it with the `check` methods for property based testing. Maybe `runLog`.